docs/en/engines/table-engines/mergetree-family/invertedindexes.md

```diff
@@ -74,7 +74,106 @@ controls the amount of data read consumed from the underlying column before a ne
 intermediate memory consumption for index construction but also improves lookup performance since fewer segments need to be checked on
 average to evaluate a query.
 
+## Full-text search of the Hacker News dataset
+
+Let's look at the performance improvements of inverted indexes on a large dataset with lots of text. We will use 28.7M rows of comments on the popular Hacker News website. Here is the table without an inverted index:
+
+```sql
+CREATE TABLE hackernews (
+    id UInt64,
+    deleted UInt8,
+    type String,
+    author String,
+    timestamp DateTime,
+    comment String,
+	dead UInt8,
+	parent UInt64,
+	poll UInt64,
+    children Array(UInt32),
+    url String,
+    score UInt32,
+    title String,
+    parts Array(UInt32),
+    descendants UInt32
+)
+ENGINE = MergeTree
+ORDER BY (type, author);
+```
+
+The 28.7M rows are in a Parquet file in S3 - let's insert them into the `hackernews` table:
+
+```sql
+INSERT INTO hackernews
+	SELECT * FROM s3Cluster(
+        'default',
+        'https://datasets-documentation.s3.eu-west-3.amazonaws.com/hackernews/hacknernews.parquet',
+        'Parquet',
+        '
+    id UInt64,
```

Comment:
Doesn't it work without the explicit structure definition?

Reply:
I tried. I get this error:
```
DB::Exception: Cannot extract table structure from Parquet format file, because there are no files with provided path in S3. 
You must specify table structure manually: Cannot extract table structure from Parquet format file. 
You can specify the structure manually. (CANNOT_EXTRACT_TABLE_STRUCTURE)
```